src/mongod_helpers.py

```diff
@@ -68,22 +74,41 @@ def check_replica_status(self) -> str:
         # look for our ip address in set of members
         for member in status["members"]:
             # get member ip without ":PORT"
-            if self._calling_unit_ip == member["name"].split(":")[0]:
+            logger.debug(member["name"].split(":")[0])
+            if replica_ip == member["name"].split(":")[0]:
                 return member["stateStr"]
 
         return None
 
-    def is_replica_ready(self) -> bool:
+    def all_replicas_ready(self) -> bool:
+        """Returns true if all replica hosts are ready."""
+        for unit_ip in self._replica_set_hosts:
+            if not self.is_replica_ready(unit_ip):
+                logger.debug("unit: %s is not ready", unit_ip)
```

Comment:
Would it be useful to log all of the unready units?

Reply:
Great catch Will! :D 

🤔 It would, but it might end up using more time / resources since we wouldn't return right away. (still `O(n)` with or without the return) so its wouldn't be a significant use of resources.

For clean-ness of code I think I will still opt for this solution unless you feel strongly :)